yt_dlp/extractor/neteasemusic.py

```diff
@@ -511,38 +564,30 @@ class NetEaseMusicDjRadioIE(NetEaseMusicBaseIE):
         'info_dict': {
             'id': '42',
             'title': '声音蔓延',
-            'description': 'md5:766220985cbd16fdd552f64c578a6b15'
+            'description': 'md5:c7381ebd7989f9f367668a5aee7d5f08'
         },
         'playlist_mincount': 40,
-        'skip': 'Blocked outside Mainland China',
     }
     _PAGE_SIZE = 1000
 
     def _real_extract(self, url):
         dj_id = self._match_id(url)
 
-        name = None
-        desc = None
+        metainfo = {}
         entries = []
         for offset in itertools.count(start=0, step=self._PAGE_SIZE):
             info = self.query_api(
-                'dj/program/byradio?asc=false&limit=%d&radioId=%s&offset=%d'
-                % (self._PAGE_SIZE, dj_id, offset),
-                dj_id, 'Downloading dj programs - %d' % offset)
-
-            entries.extend([
-                self.url_result(
-                    'http://music.163.com/#/program?id=%s' % program['id'],
-                    'NetEaseMusicProgram', program['id'])
-                for program in info['programs']
-            ])
-
-            if name is None:
-                radio = info['programs'][0]['radio']
-                name = radio['name']
-                desc = radio['desc']
+                f'dj/program/byradio?asc=false&limit={self._PAGE_SIZE}&radioId={dj_id}&offset={offset}',
+                dj_id, note=f'Downloading dj programs - {offset}')
+
+            entries.extend(self._get_entries(info, 'programs'))
+            if not metainfo:
+                metainfo = traverse_obj(info, ('programs', 0, 'radio', {
+                    'title': ('name', {str}),
+                    'description': ('desc', {str}),
+                }))
 
             if not info['more']:
                 break
 
-        return self.playlist_result(entries, dj_id, name, desc)
+        return self.playlist_result(entries, dj_id, **metainfo)
```

Comment:
instead of `.extend` loop, we could convert this into a generator
```py
    def _real_extract(self, url):
        dj_id = self._match_id(url)
        metainfo = {}

        def entries():
            for offset in itertools.count(start=0, step=self._PAGE_SIZE):
                info = self.query_api(
                    f'dj/program/byradio?asc=false&limit={self._PAGE_SIZE}&radioId={dj_id}&offset={offset}',
                    dj_id, note=f'Downloading dj programs - {offset}')

                yield from self._get_entries(info, 'programs')

                if not metainfo:
                    metainfo.update(traverse_obj(info, ('programs', 0, 'radio', {
                        'title': ('name', {str}),
                        'description': ('desc', {str}),
                    })))

                if not info['more']:
                    break

        return self.playlist_result(entries(), dj_id, **metainfo)
```

Reply:
Problem with generator is that metainfo is not extracted until the `entries()` get evaluated, meaning the returned value will always get `self.playlist_result(entries(), dj_id, **{})`.

Reply:
Ah my bad, I didn't think about `metainfo` problem, you can ignore this suggestion then. And you don't need to use `_get_entries` here if it's a problem

Reply:
It should be good now: https://github.com/yt-dlp/yt-dlp/pull/8181/commits/66442f1ba8f8d5fb6a924937369d16cf4cbae24c